lib/pf/web/device_registration.pm

```diff
@@ -18,65 +18,95 @@ use warnings;
 use HTML::Entities;
 use Readonly;
 
-use pf::config;
-use pf::file_paths qw($allowed_device_oui_file $allowed_device_types_file);
+use pf::constants;
+use pf::config qw(%Config);
 use pf::enforcement qw(reevaluate_access);
 use pf::node qw(node_register is_max_reg_nodes_reached);
 use pf::util;
+use pf::error qw(is_success);
+use pf::log;
 use pf::web;
 use pf::web::custom;    # called last to allow redefinitions
 
 use pf::authentication;
 use pf::Authentication::constants;
 use List::MoreUtils qw(any);
 
-Readonly our @DEVICE_OUI => _load_file_into_array($allowed_device_oui_file);
-Readonly our @DEVICE_TYPES => _load_file_into_array($allowed_device_types_file);
-
 =head1 SUBROUTINES
 
-=over
+=cut
+
+=item mac_vendor_id
+
+Get the matching mac_vendor_id from Fingerbank
 
 =cut
 
+sub mac_vendor_id {
+    my ($mac) = @_; 
+    my $logger = get_logger();
+
+    my ($status, $result) = fingerbank::Model::MAC_Vendor->find([{ mac => $mac }, {columns => ['id']}]);
+
+    if(is_success($status)){
+        return $result->id;
+    }else {
+        $logger->debug("Cannot find mac vendor ".$mac." in the database");
+    }
+}
 
-=item _load_file_into_array
+=item device_id
 
-Loads each line of file into array
-Trimming spaces and removing shell style comments from each line
+Get the matching device_id from Fingerbank
 
 =cut
 
-sub _load_file_into_array {
-    my ($file_name) = @_;
-    my @items;
-    if(-r $file_name) {
-        local *FILE;
-        open(FILE,$file_name);
-        @items =
-            grep {$_}
-            map {
-                #Getting rid of newlines, comments and trimming spaces
-                chomp; s/#.*$//; s/^\s+//; s/\s+$//;
-                $_;
-            } <FILE>;
-        close(FILE);
+sub device_id {
+    my ($mac_vendor_id) = @_; 
+    my $logger = get_logger();
+
+    my ($status, $result) = fingerbank::Model::Combination->find([{ mac_vendor_id => $mac_vendor_id }, {columns => ['device_id']}]);
```

Comment:
Here it seems you're just grabing the first combination that has the mac vendor you want.

What you should do instead is find the most popular device ID for the MAC vendor you provide, not just any of them

So replicating the logic from this query:
select device_id from combination where mac_vendor_id=$mac_vendor_id group by device_id order by count(device_id) desc limit 1;